apps/music/js/ui/views/tiles_view.js

```diff
@@ -120,22 +122,25 @@ var TilesView = {
       artistName.classList.add('sub-tile-title');
     }
 
-    var NUM_INITIALLY_VISIBLE_TILES = 8;
-    var INITIALLY_HIDDEN_TILE_WAIT_TIME_MS = 1000;
-
-    var setTileBackgroundClosure = function(url) {
-      tile.style.backgroundImage = 'url(' + url + ')';
-    };
-
-    if (this.index <= NUM_INITIALLY_VISIBLE_TILES) {
-      // Load this tile's background now, because it's visible.
-      AlbumArtCache.getCoverURL(result).then(setTileBackgroundClosure);
-    } else {
-      // Defer loading hidden tiles until the visible ones are done.
-      setTimeout(function() {
-        AlbumArtCache.getCoverURL(result).then(setTileBackgroundClosure);
-      }, INITIALLY_HIDDEN_TILE_WAIT_TIME_MS);
-    }
+    var index = this.index;
+    LazyLoader.load('js/metadata/album_art_cache.js', function() {
+      var NUM_INITIALLY_VISIBLE_TILES = 8;
+      var INITIALLY_HIDDEN_TILE_WAIT_TIME_MS = 1000;
+
+      var setTileBackgroundClosure = function(url) {
+        tile.style.backgroundImage = 'url(' + url + ')';
+      };
+
+      if (index <= NUM_INITIALLY_VISIBLE_TILES) {
+        // Load this tile's background now, because it's visible.
+        AlbumArtCache.getThumbnailURL(result).then(setTileBackgroundClosure);
+      } else {
+        // Defer loading hidden tiles until the visible ones are done.
+        setTimeout(function() {
+          AlbumArtCache.getThumbnailURL(result).then(setTileBackgroundClosure);
+        }, INITIALLY_HIDDEN_TILE_WAIT_TIME_MS);
```

Comment:
I had never seen this code before... it is probably a good optimization, but yuck!  Every tile after the first 8 is delayed by a second even after the first page is fully loaded.  This is a problem that might just go away if the albums were enumerated with enumerate() instead of enumerateAll()